lib/web_apis/api_compat_data.es6.js

```diff
@@ -120,3 +120,96 @@ foam.CLASS({
     },
   ],
 });
+
+foam.CLASS({
+  name: 'CompatClassGenerator',
+  package: 'org.chromium.apis.web',
+
+  documentation: `Singleton that encapsulates logic for generating a compat
+      data class based on a collection of known releases.`,
+
+  axioms: [foam.pattern.Singleton.create()],
+
+  methods: [
+    {
+      name: 'generateSpec',
+      documentation: `Generate the "spec" that would be passed to foam.CLASS()
+          to declare a class of given [pkg].[name], based on given releases.`,
+      code: function(pkg, name, releases) {
+        const generator = this;
+        let spec = {
+          class: 'Model',
+          package: pkg,
+          name: name,
+          extends: 'org.chromium.apis.web.AbstractApiCompatData',
+
+          requires: ['org.chromium.apis.web.CompatProperty'],
+
+          properties: [],
+        };
+
+        let propSpecs = releases.map(release => {
+          return {
+            class: 'org.chromium.apis.web.CompatProperty',
+            name: this.propertyNameFromRelease(release),
+            label: this.propertyLabelFromRelease(release),
+            release,
+          };
+        });
+        spec.properties = spec.properties.concat(propSpecs);
+
+        return spec;
+      },
+    },
+    {
+      name: 'generateClass',
+      documentation: `Generate a class based on a spec or foam.core.Model. The
+          former is used during data generation phase; the latter is used when
+          loading a model for existing data.`,
+      code: function(specOrModel) {
+        if (foam.core.Model.isInstance(specOrModel)) {
+          // Run post-model instantiation steps from:
+          // https://github.com/foam-framework/foam2/blob/632c6467c0a7f123a2270cc549db5d8e3d93b574/src/foam/core/Boot.js#L201
+          const model = specOrModel;
+          model.validate();
+          const cls = model.buildClass();
+          cls.validate();
+          foam.register(cls);
+          foam.package.registerClass(cls);
+          return cls;
+        } else {
+          // Declare class from spec as usual.
+          const spec = specOrModel;
+          foam.CLASS(spec);
+          return foam.lookup(`${spec.package}.${spec.name}`);
+        }
+      },
+    },
+    {
+      name: 'propertyNameFromRelease',
+      documentation: `Produce a consistent camelCase property name from a
+          release object. The property is used to store compat data for the
+          release.`,
+      code: function (release) {
+        return release.browserName.charAt(0).toLowerCase() +
+            release.browserName.substr(1).replace(/[^a-z]/ig, '_') +
```

Comment:
Might be easier to do the concats first, then sanitize the whole string. I don't think stripping numbers out of browser and OS name is worth the extra regexp passes.